core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/entity/player/input/BedrockMovePlayer.java

```diff
@@ -87,24 +87,29 @@ static void translate(GeyserSession session, PlayerAuthInputPacket packet) {
             session.setLookBackScheduledFuture(null);
         }
 
-        // Simulate jumping since it happened this tick, not from the last tick end.
-        if (entity.isOnGround() && packet.getInputData().contains(PlayerAuthInputData.START_JUMPING)) {
-            entity.setLastTickEndVelocity(Vector3f.from(entity.getLastTickEndVelocity().getX(), Math.max(entity.getLastTickEndVelocity().getY(), entity.getJumpVelocity()), entity.getLastTickEndVelocity().getZ()));
-        }
-
-        // Due to how ladder works on Bedrock, we won't get climbing velocity from tick end unless if you're colliding horizontally. So we account for it ourselves.
-        boolean onClimbableBlock = session.getTagCache().is(BlockTag.CLIMBABLE, session.getGeyser().getWorldManager().blockAt(session, entity.getPosition().sub(0, EntityDefinitions.PLAYER.offset(), 0).toInt()).block());
-        if (onClimbableBlock && packet.getInputData().contains(PlayerAuthInputData.JUMPING)) {
-            entity.setLastTickEndVelocity(Vector3f.from(entity.getLastTickEndVelocity().getX(), 0.2F, entity.getLastTickEndVelocity().getZ()));
-        }
+        boolean verticalCollision = packet.getInputData().contains(PlayerAuthInputData.VERTICAL_COLLISION);
 
         // Client is telling us it wants to move down, but something is blocking it from doing so.
         boolean isOnGround;
         if (hasVehicle) {
             // VERTICAL_COLLISION is not accurate while in a vehicle (as of 1.21.62)
             isOnGround = Math.abs(entity.getLastTickEndVelocity().getY()) < 0.1;
         } else {
-            isOnGround = packet.getInputData().contains(PlayerAuthInputData.VERTICAL_COLLISION) && entity.getLastTickEndVelocity().getY() < 0;
+            isOnGround = verticalCollision && entity.getLastTickEndVelocity().getY() < 0;
+
+            // We only have to check for these cases if player is having vertical collision else the onGround status will always be false.
+            // These cases will always (I think so?) have upwards motion so don't even bother checking what the actual tick end y, the status is going to be false.
+            if (verticalCollision) {
+                if (entity.isOnGround() && packet.getInputData().contains(PlayerAuthInputData.START_JUMPING)) {
+                    isOnGround = false;
+                }
+
+                // Due to how ladder works on Bedrock, we won't get climbing velocity from tick end unless if you're colliding horizontally. So we account for it ourselves.
+                boolean onClimbableBlock = session.getTagCache().is(BlockTag.CLIMBABLE, session.getGeyser().getWorldManager().blockAt(session, entity.getPosition().sub(0, EntityDefinitions.PLAYER.offset(), 0).toInt()).block());
```

Comment:
SessionPlayerEntity (entity var here) has a climbable check that can be used, that one should be more correct as well